tests/e2e/mod.rs

```diff
@@ -19,60 +20,74 @@ use tracing::{info, warn};
 
 mod dev;
 mod subgraph;
+mod supergraph;
 
 const GRAPHQL_TIMEOUT_DURATION: Duration = Duration::from_secs(30);
 
 #[derive(Debug, Deserialize)]
-struct ReducedSupergraphConfig {
+pub struct RetailSupergraph {
     subgraphs: HashMap<String, ReducedSubgraphConfig>,
+    working_dir: Option<String>,
```

Comment:
I don't think we can just use strings here because I'm pretty sure that when the TempDir goes out scope the underlying directory gets deleted as well. Could we use a solution like this: https://andrewra.dev/2019/03/01/testing-in-rust-temporary-files/#a-reasonable-solution

Where we keep the TempDir object as a private field on the struct so that we can maintain ownership of it but keep the other nice interface?